# BINF-6110_Assignment-1: Assembling and alignment of the _Salmonella enterica_ genome using long read data sequenced from an Oxford Nanopore Sequencer
### Evelyn Hoover
## Assignment 1 Part 1: Introduction and Proposed Methods
## Introduction  
 ### Biological Context  
  Bacteria are a phylum of prokaryotic microorganisms that are abundant worldwide. Most species within this phylum are harmless or even beneficial to humans and pose no threat to our health or wellbeing. However, there are some species that can enter or otherwise come into contact with a human and cause a wide range of harmful symptoms. _Salmonella spp._ are generally well-known for this, commonly in conjunction with raw chicken products. _Salmonella enterica_ specifically can cause gastroenteritis, bacteremia, or enteric fever depending on how the bacteria enters the body and the severity of infection (Public Health Agency of Canada). _S. enterica_ is a gram-negative faculatative anaerobe and spread of infection is facilitated by contact with infected animals or humans, their feces, or consumption of infected food or water. Infections can become fatal if left untreated, especially those of the population that are more vulnerable, such as children and the elderly. For this reason, it is important to monitor spread of infections. 
  Antibiotics are commonly used to treat _S. enterica_ infections. For this reason, populations expressing antibiotic resistance are often of concern because of their genetic ability to resist treatment and pass this resistant ability to more cells in the population (_Antimicrobial Resistance_, WHO). If left unchecked this can render antibiotic treatments ineffective and reduce the ability to treat infections, leading to more severe cases of infection (_Antimicrobial Resistance_, WHO). Therefore, sequencing samples of _S. enterica_ and aligning them to reference genomes to monitor for mutations that could give rise to antibiotic resistance is essential for minimising this threat.
 ### Goals  
  The goal of this assignment is to assemble the long read sequences of  _S. enterica_ produced from a Nanopore sequencer and align the assembled genome to a reference genome from the literature. In a real-world scenario this technique would often be used to compare the genetics of a sample to recorded literature to determine mutations or the effects of knockouts of genes within the population. In this circumstance, the goal is to successfully clean the data, assemble it, align it, and visualise it.
 ### Challenges and Limitations  
  Considering the sequences to be used are long reads there may be more errors than if short read sequencing were used. For this reason steps may need to be repeated to ensure the sequences are the highest quality possible before moving onto the next step. However, long read sequencing is necessary when looking for mutations and comparing the genome to a reference. Additionally, as someone with very little experience in bioinformatics as a whole, the progression of this project will not be nearly as linear as it could be with someone with more experience. However, that is simply what happens when learning a new skill.
## Methods  
 ### Data Description  
  The data used in this assignment is a long read sequence from an Oxford Nanopore Sequencer. The sequence was generated using R10 chemistry and the reads have an expected accuracy of Q20+ and a read length distribution of N50:5-15kb. The data is provided as a FASTQ file.
 ### Quality Control
  The FASTQ file is initially run through FASTQC which is generally used for short read sequences but can also be used to run long read sequences in Linux (_LongReadSum: A Fast and Flexible Quality Control and Signal Summarization Tool for Long-Read Sequencing Data_, 2025). This process works to trim the data of any errors so that it can be assembled properly. While FASTQC can be used on long read sequences, it does have some pitfalls in that it does not provide metrics important for long reads nor read and base QC mapping (_LongReadSum: A Fast and Flexible Quality Control and Signal Summarization Tool for Long-Read Sequencing Data_, 2025). However, FASTQC is simple to use for new users and can provide results sufficient for this project.
 ### Assembly
  To perform de novo assembly on the reads to form a consensus genome, Flye will be used. Flye is optimal for this process as it constructs an assembly graph using disjointed segments which allows it to improve assembly at long and repetetive regions (Wang et al.). This process does not require manual intervention, and a polishing step was not required.
 ### Alignment  
  After assembly, the consensus genome will be aligned to a reference genome. The reference genome FASTA file will be obtained from NCBI (Salmonella Enterica Subsp. Enterica Serovar Typhimurium Str. LT2, Comp - Nucleotide - NCBI). minimap2 will be used to align the reads to the reference genome. A CIGAR will be generated and the outputs will be formed into the SAM format.
 ### Visualisation  
  To visualise the alignment it will first have to be sorted and indexed. For this samtools will be used to convert the file to a BAM file and that file will then be sorted and indexed using samtools. Visualisation will then be done using IGV. Proksee will be used to generate a whole-genome view.  
## Results  
 <img width="841" height="422" alt="senterica_fullgenome" src="https://github.com/user-attachments/assets/41b6c46d-a27c-460c-89b4-93985fd75dd0" />  \
 <ins> Figure 1. </ins> A full genome view of the *Salmonella enterica* assembled genome visualized using Proksee. The three contigs of the genome are demarkated in different shades of gray at the centre of the structure with lines indicating functionality of the associated genes in different colours stemming from the centre. The bakta tool within Proksee was used to determine the gene functionalities. 
<img width="1281" height="567" alt="contig4_variantcalling" src="https://github.com/user-attachments/assets/9cde0e36-41c9-413a-b1d0-863bebbdb137" />  
<ins> Figure 2. </ins> IGV plot demonstrating variant calling on contig 4 from the assembled *Salmonella enterica* genome when aligned to the reference genome. This contig was relatively short and had fairly high levels of variance, especially toward the end of the contig.  
A) <img width="1281" height="567" alt="senterica_SNP3" src="https://github.com/user-attachments/assets/6f1b7815-c665-4c85-b527-3454bdd194bc" />
B) <img width="1281" height="567" alt="senterica_SNP1" src="https://github.com/user-attachments/assets/e8116857-30f3-4d86-9f8d-7a1f0cf47974" />
C) <img width="1281" height="567" alt="senterica_SNP2" src="https://github.com/user-attachments/assets/18d16366-d8aa-4042-b56a-e7c2da015583" />  
<ins> Figure 3 </ins> Three examples of SNPs from the variant calling visualised on IGV. The first image (A) is two SNPs on contig 3. The second two images (B and C) are of a collection of SNPs seen on contig 4. B) is on an earlier section of contig 4 while C) is on a later section.  
## Discussion  
 After assembling the genome and performing variant calling of it against the reference genome it is observed that there are many variant genes across the genome. Notably, there is an observed high level of variance in contig 4. From the information gathered by the bakta tool in Proksee, it is determined that this is where the origin of transfer and origin of replication sites are located. Variants in these regions suggest mutations which could influence how the cells replicate, in ways that could be good or bad for the survival of the colony. For instance, mutations in this region could influence the antimicrobial resistance of these cells, possibly changing the antibiotic that would be used against infection (Watterford & Warrington, 2023). However, with how many variants there are in this region, it is also possible that these cells could have mutated in the region enough to not be able to replicate further (Watterford & Warrington, 2023). The outcome of these variants is fully dependent on the observed nature of the colony.  
 Variations in the origin of replication region can induce genomic structural variation in which the order of DNA genome sections can be altered which can then involve multiple genes in this variation (Waters *et al*., 2025). This could potentially explain the high degree of variation observed throughout the genome. All together, these changes will likely influence various transcription processes which will wholely change the way the cell is able to function.
## Conclusion  
 All in all, this methodology can be used successfully for assembling a long read bacterial ONT sequence and comparing that sequence to a reference genome, while also being simple enough for a novice to learn from. While there are limitations and potential pitfalls in this process, it should be reliable enough to perform the project sufficiently. Following the genome assembly and variant calling, it is determined that there are likely various mutations in the origin of replication and origin of transfer regions of the genome which lead to genomic structural variation within the rest of the genome which influences higher levels of variation throughout the genome. This is likely to influence the way the cell is able to function and how the colony replicates as a whole, likely promoting antimicrobial resistance.
## References  
*Antimicrobial Resistance*. https://www.who.int/news-room/fact-sheets/detail/antimicrobial-resistance. 
*LongReadSum: A Fast and Flexible Quality Control and Signal Summarization Tool for Long-Read Sequencing Data*. (2025) Computational and Structural Biotechnology Journal, vol. 27, pp. 556–63.  
*Pathogen Safety Data Sheets: Infectious Substances – Salmonella Enterica Spp.* (2001) Public Health Agency of Canada. https://www.canada.ca/en/public-health/services/laboratory-biosafety-biosecurity/pathogen-safety-data-sheets-risk-assessment/salmonella-enterica.html.  
*Salmonella Enterica Subsp. Enterica Serovar Typhimurium Str. LT2, Comp - Nucleotide*. - NCBI https://www.ncbi.nlm.nih.gov/nuccore/NC_003197.2  
Wang, Yunhao, *et al*. (2021) *Nanopore Sequencing Technology, Bioinformatics and Applications*. Nature Biotechnology, vol. 39, no. 11, pp. 1348–65.  
Waters, E.V. *et al*., (2025) *Bacterial genome structural variation: prevalence, mechanisms, and consequences*. Trends in Microbio. vol. 33, no. 8, p875-886.  
Wattford, S. & Warrington, S.J. (2023) *Bacterial DNA Mutations* NIH. https://www.ncbi.nlm.nih.gov/books/NBK459274/https://www.ncbi.nlm.nih.gov/books/NBK459274/
